src/FSharpPlus/Extensions.fs

```diff
@@ -295,6 +295,103 @@ module String =
             |> String.filter (fun ch -> CharUnicodeInfo.GetUnicodeCategory ch <> UnicodeCategory.NonSpacingMark)
             |> normalize NormalizationForm.FormC
 
+    /// Pads the beginning of the given string with spaces so that it has a specified total length.
+    let padLeft totalLength (source: string) = source.PadLeft totalLength
+    /// Pads the beginning of the given string with a specified character so that it has a specified total length.
+    let padLeftWith totalLength paddingChar (source: string) = source.PadLeft (totalLength, paddingChar)
+    /// Pads the end of the given string with spaces so that it has a specified total length.
+    let padRight totalLength (source: string) = source.PadRight totalLength
+    /// Pads the end of the given string with a specified character so that it has a specified total length.
+    let padRightWith totalLength paddingChar (source: string) = source.PadRight (totalLength, paddingChar)
+
+    /// Removes all leading and trailing occurrences of specified characters from the given string.
+    let trim      (trimChars: char seq) (source: string) = source.Trim (Seq.toArray trimChars)
```

Comment:
I'm a bit unsure whether the `trimChars` param should be a `char seq`, a `char array` or a `string`. Any arguments for or against any of them?

Reply:
I think `char seq` is the best choice here as it can receive `char list`, `Set<char>`, etc

Reply:
Yes, that's a good argument for `char seq`.

As for `char []` I would say it's a bit more performant, anyway the list is expected to be short.

For `string` it's interesting to note that it matches the collection type. Here I'm more thinking again in generic terms, what if we decide in the future to create a trim for lists, arrays ...

Reply:
When generalized, `trim` can be thought as a variant of `except`. It takes `'a seq` as elements to exclude regardless of collection type. I think it would make more sense to follow the same design.